src/test/test-strv.c

```diff
@@ -656,6 +657,8 @@ static void test_strv_make_nulstr_one(char **l) {
         assert_se(strv_make_nulstr(q, &c, &m) >= 0);
         assert_se(m == n);
         assert_se(memcmp(b, c, m) == 0);
+
+        NULSTR_FOREACH(s, b);
```

Comment:
In the loop, please compare `*s` with each of the corresponding strings in `l`. And also check that all items in `l` are covered. This way the test will be more likely to trigger failure if `b` is buggy.